examples/ASMapNode/Sample/MapHandlerNode.m

```diff
@@ -90,6 +91,22 @@ - (void)didLoad
   [_liveMapToggleButton setTitle:[self liveMapStr] withFont:nil withColor:[UIColor blueColor] forState:ASControlStateNormal];
   [_liveMapToggleButton setTitle:[self liveMapStr] withFont:[UIFont systemFontOfSize:14] withColor:[UIColor blueColor] forState:ASControlStateHighlighted];
   [_liveMapToggleButton addTarget:self action:@selector(toggleLiveMap) forControlEvents:ASControlNodeEventTouchUpInside];
+  
+  // avoiding retain cycles
+  __weak MapHandlerNode *weakSelf = self;
+  
+  self.mapNode.imageForStaticMapAnnotationBlock = ^UIImage *(id<MKAnnotation> annotation, CGPoint *centerOffset){
+    MapHandlerNode *grabbedSelf = weakSelf;
+    if (grabbedSelf) {
+      if ([annotation isKindOfClass:[CustomMapAnnotation class]]) {
+        MKAnnotationView *av = [grabbedSelf annotationViewForAnnotation:annotation];
+        return av.image;
```

Comment:
We can't allocate views in this block, since we're off the main thread. @michalziman Since CustomMapAnnotation has an image property, can you modify this like:

```objectivec
CustomMapAnnotation *customAnnotation = (CustomMapAnnotation *)annotation;
return customAnnotation.image
```